test/common/authdelay_first_broker_login.go

```diff
@@ -177,7 +203,38 @@ func loginToThreeScale(t *testing.T, tsHost, username, password string, idp stri
 			})
 		})
 
-		t.Logf("browser URL %s", browser.Url().RequestURI())
+		// checks if an error happened in the login
+		if browser.StatusCode() == 502 {
```

Comment:
I am wondering about this part.
a) how did you find out a 502 status code was returned?
b) any chance we can call some function instead of the code below? It seems like something that's probably already implemented elsewhere.

Reply:
Just to give a bit of context on this, when I created the e2e test for the authentication delay plugin I was verifying the authentication of new users, but other tests are using the `loginToThreeScale` function for existing users and as a few tests are using the same user to login to 3scale and this user is already authenticated the test get stuck at an endpoint like the one below 

```/auth/realms/openshift/broker/openshift-v4/endpoint?code=5eoXNSKQ6Fw_uvcYiEsemDU6EOUE90KG0V9Me6PPblQ&state=ZWlDcGsbqD00QL_lVQ1AIKM1MAgn_T06tRNIs5EvMbY.kBoSCwUdIwM.3scale```

a) I logged in to 3scale and got the link with the endpoint that was similar to the one that was getting stuck in the test and it returned 502.

b) I suppose you are right,  some parts of this code is kind repeated and I could actually create some functions, do you reckon it would be used by other tests?

Reply:
Thank you for the explanation.

As for b), probably not. Let's leave it as is.